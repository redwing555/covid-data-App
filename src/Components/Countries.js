import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Country from './Country';
import Query from './Query';

const Countries = ({
  countries, total, loading, handleClick,
}) => {
  const [filtered, setFiltered] = useState(Object.keys(countries));

  const filterCountries = (expression) => Object.keys(countries).filter((country) => {
    const regex = new RegExp(expression, 'gi');
    return country.match(regex);
  });

  const handleChange = (event) => setFiltered(filterCountries(event.target.value));

  useEffect(() => {
    setFiltered(Object.keys(countries));
  }, [countries]);

  return (
    <div>
      <span>
        {' '}
        {total && Intl.NumberFormat('de-DE').format(total.today_confirmed) }
      </span>
      <Query handleChange={handleChange} />
      {loading && <span>...loading</span> }
      <ul>
        {filtered && filtered.map((key) => (
          (key !== 'Kosovo')
              && (
              <li key={key}>
                <Country
                  country={key}
                  totalConfirmed={countries[key].today_confirmed}
                  handleClick={handleClick}
                />
              </li>
              )
        ))}
      </ul>
      {!filterCountries.length && !loading && <span>Sorry</span>}
    </div>
  );
};

Countries.defaultProps = {
  loading: true,
  total: {},
};

Countries.propTypes = {
  countries: PropTypes.objectOf(PropTypes.object).isRequired,
  total: PropTypes.shape({
    today_confirmed: PropTypes.number,
  }),
  handleClick: PropTypes.func.isRequired,
  loading: PropTypes.bool,

};

export default Countries;
