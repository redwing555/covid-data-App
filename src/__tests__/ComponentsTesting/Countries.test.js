import React from 'react';
import renderer from 'react-test-renderer';
import Countries from '../../Components/Countries';

const mockedData = {
  Morocco: {
    new_cases: 485,
    death: 4,
    today_confirmed: 5645,
  },
  Algeria: {
    new_cases: 45,
    death: 5,
    today_confirmed: 545,
  },
};

const mockedTotalCases = {
  today_confirmed: 33868,
  today_new_confirmed: 6753375,
  today_deaths: 575238,
};

describe('Countries', () => {
  it('renders ! ', () => {
    const tree = renderer
      .create(
        <Countries
          countries={mockedData}
          total={mockedTotalCases}
          loading={false}
          handleClick={() => 7 + 5}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
