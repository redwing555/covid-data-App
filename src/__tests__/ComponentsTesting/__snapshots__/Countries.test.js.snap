// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders !  1`] = `
<div>
  <div
    className="header-wrapper"
  >
    <div
      className="world-header"
    >
      <img
        alt="world map"
        className="world-map"
        src="/worldmap.svg"
      />
    </div>
    <div
      className="world-info"
    >
      <span
        className="world-cases"
      >
        The world has 
      </span>
      <span>
         
        33.868
      </span>
      <span
        className="world-cases"
      >
        Coronavirus Cases 
      </span>
    </div>
  </div>
  <form
    className="form"
  >
    <span
      className="search-by"
    >
      Stats by Country
    </span>
    <input
      className="input-search"
      onChange={[Function]}
      placeholder="Enter country name"
      type="text"
    />
  </form>
  <ul
    className="countries-list"
  >
    <li
      className="countries"
    >
      <button
        className="country"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="map"
          className="country-map"
          height="100"
          src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ma/vector.svg"
          width="100"
        />
        <div
          className="country-data"
        >
          <svg
            className="info-icon"
            color="white"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
            />
            <path
              d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
            />
          </svg>
          <h1
            className="country-name"
          >
            MOROCCO
          </h1>
          <span
            className="total-confirmed"
          >
            5.645
          </span>
        </div>
      </button>
    </li>
    <li
      className="countries"
    >
      <button
        className="country"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="map"
          className="country-map"
          height="100"
          src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/dz/vector.svg"
          width="100"
        />
        <div
          className="country-data"
        >
          <svg
            className="info-icon"
            color="white"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
            />
            <path
              d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
            />
          </svg>
          <h1
            className="country-name"
          >
            ALGERIA
          </h1>
          <span
            className="total-confirmed"
          >
            545
          </span>
        </div>
      </button>
    </li>
  </ul>
</div>
`;
